azurelinuxagent/common/protocol/extensions_goal_state_from_extensions_config.py

```diff
@@ -61,9 +61,17 @@ def _parse_extensions_config(self, xml_text, wire_client):
         for ga_family in ga_families:
             name = findtext(ga_family, "Name")
             version = findtext(ga_family, "Version")
+            is_version_from_rsm = findtext(ga_family, "IsVersionFromRSM")
+            is_vm_enabled_for_rsm_upgrades = findtext(ga_family, "IsVMEnabledForRSMUpgrades")
             uris_list = find(ga_family, "Uris")
             uris = findall(uris_list, "Uri")
-            family = VMAgentFamily(name, version)
+            family = VMAgentFamily(name)
+            if version is not None:
```

Comment:
why do we need these checks for None, if VMAgentFamily initializes these properties to None anyways?

this gives the impresion that those properties should not be None